pkg/registration/spoke/event_handler.go

```diff
@@ -0,0 +1,68 @@
+package spoke
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	corev1 "k8s.io/api/core/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+)
+
+type bootstrapKubeconfigEventHandler struct {
+	bootstrapKubeconfigSecretName *string
+	cancel                        context.CancelFunc
+}
+
+func (hc *bootstrapKubeconfigEventHandler) Name() string {
+	return "bootstrap-hub-kubeconfig"
+}
+
+// implement cache.ResourceEventHandler.OnAdd
+func (hc *bootstrapKubeconfigEventHandler) OnAdd(obj interface{}, isInInitialList bool) {
+}
+
+// implement cache.ResourceEventHandler.OnUpdate
+func (hc *bootstrapKubeconfigEventHandler) OnUpdate(oldObj, newObj interface{}) {
+	newSecret, ok := newObj.(*corev1.Secret)
+	if !ok {
+		utilruntime.HandleError(fmt.Errorf("invalid secret object: %v", newObj))
+		return
+	}
+
+	if newSecret.Name != *hc.bootstrapKubeconfigSecretName {
+		return
+	}
+
+	oldSecret, ok := oldObj.(*corev1.Secret)
+	if !ok {
+		utilruntime.HandleError(fmt.Errorf("invalid secret object: %v", oldObj))
+		return
+	}
+
+	if !reflect.DeepEqual(newSecret.Data, oldSecret.Data) {
+		// Restart immediately if the bootstrap kubeconfig changes. Otherwise, in the backup restore scenario,
```

Comment:
backup restore scenario -> switch hub scenario
the restored one -> the new one

Reply:
Comments updated.